ui/src/helpers/api.js

```diff
@@ -46,6 +46,99 @@ export default class Api {
         return response.data;
     }
 
+    //#region User api
+
+    getUsers = async(filter = "") =>{
+        const response = await this.client.get(`/user/listUsers?usernameFilter=${filter}`);
+        return response.data;
+    }
+
+    getUserDetails = async(user) => {
+        const response = await this.client.get(`/user/${user.userId}`);
+        return response.data;
+    }
+
+    updateUser = async(user) => {
```

Comment:
if we have a model for user let's use it here as async(user: User) => ..

Reply:
its a js file, not TS.
I was glad to to use TYPES

Reply:
oh you're right
if you want you can change the file to ts, we can do it as a different task though
feel free to ignore all my comments regarding types in this file if you decide to skip it for now